src/main/java/jp/co/soramitsu/sora/json/Flattener.java

```diff
@@ -0,0 +1,172 @@
+package jp.co.soramitsu.sora.json;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.val;
+import lombok.var;
+
+
+public class Flattener {
+
+  private int features = 0;
+  private char dictkey = '/';
+  private char arrkey = '_';
+  private ObjectMapper mapper = new ObjectMapper();
+
+  private void flatten(JsonNode root, String path, ObjectNode out) {
+    if (root.isObject()) {
+      if (root.size() == 0) {
+        // it is empty object
+        out.set(path, root);
+        return;
+      }
+
+      val fields = root.fields();
+      while (fields.hasNext()) {
+        var it = fields.next();
+        var key = sanitize(it.getKey());
+
+        flatten(it.getValue(), path + dictkey + key, out);
+      }
+    } else if (root.isArray()) {
+      if (root.size() == 0) {
+        // it is empty array
+        out.set(path, root);
+        return;
+      }
+
+      val fields = root.iterator();
+      long counter = 0;
+      while (fields.hasNext()) {
```

Comment:
I suggest to refactor using `Stream`, this will allow to clean code from iterators, counters and other imperative code

Reply:
same applies to other parts of code with iteration loops

Reply:
I am new to rxjava, maybe you can suggest how to do this?